chalk-solve/src/clauses/program_clauses.rs

```diff
@@ -275,6 +275,172 @@ impl<I: Interner> ToProgramClauses<I> for StructDatum<I> {
     }
 }
 
+impl<I: Interner> ToProgramClauses<I> for FnDefDatum<I> {
```

Comment:
This seems mostly duplicated from `StructDatum`?

Reply:
Interesting stuff! We'll have to think about just which rules we want here...